mw-webapp/src/pages/authPage/AuthPage.tsx

```diff
@@ -1,14 +1,36 @@
-import {Link} from "react-router-dom";
-import {useAuth} from "src/utils/useAuth";
+import {User} from "firebase/auth";
+import {useEffect, useState} from "react";
+import {Link, useNavigate} from "react-router-dom";
+import {Button} from "src/component/button/Button";
+import {getNewUserCredentials} from "src/utils/auth/getNewUserCredentials";
+import {handleUserAuthState} from "src/utils/auth/handleAuthState";
+import {handleLogIn} from "src/utils/auth/handleLogIn";
+import {handleLogOut} from "src/utils/auth/handleLogOut";
 
 export const AuthPage = () => {
-  const handleGoogleSignIn = useAuth();
+  const [user, setUser] = useState<User | null>(null);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    return () => {
+      handleUserAuthState(setUser);
+      getNewUserCredentials(navigate, "/main");
```

Comment:
why does getNewUserCredentials returns void, and its interface include navigate and part of url as a string? Fix all this please)

Reply:
This function writes user data to database.
It includes navigate to redirect to the page with a table only if the user logged in. If the user already logged in three is no redirection to the page with the table. I write in such way because we don't have logout button on the table page.
Npw I delete navigate params from this function.

Reply:
Victor mean that if you call function get... then this function should return something so if this function doesn't return anything  need to change naming